source/mainnet/smart-contracts/guides/unit-test-contract.rst

```diff
@@ -342,8 +342,11 @@ state and balance fields:
 Reentrancy
 ----------
 
-You should watch out for *reentrancy problems*, which can occur when calls to
-``invoke_contract`` end up updating the state of your own contract.
+When invoking another smart contract, you give away control to that contract in the middle of execution.
+The external contract can, for example, call back entrypoints of your contract.
+This behavior is called *reentrancy* and is well-known from concurrency: a procedure can be interrupted in the middle of its execution, called again, and then resume execution.
+
+The state of your contract might not be the same before and after ``invoke_contract``, since the contract you call can update the state of your own contract.
```

Comment:
Can we be a bit more precise. In other places we are adamant about the fact that only you and you alone can update the contract state. But here it now says "the contract you call can update the state ..." which is technically incorrect.

Can you please rephrase this so that it is both technically correct, and fits into the narrative we try to have.